lib/web_ui/lib/src/engine/pointer_binding.dart

```diff
@@ -1133,12 +1150,14 @@ class _PointerAdapter extends _BaseAdapter with _WheelEventListenerMixin {
   }
 
   int _getPointerId(DomPointerEvent event) {
-    // We force `device: _mouseDeviceId` on mouse pointers because Wheel events
-    // might come before any PointerEvents, and since wheel events don't contain
-    // pointerId we always assign `device: _mouseDeviceId` to them.
-    final ui.PointerDeviceKind kind = _pointerTypeToDeviceKind(event.pointerType!);
-    return kind == ui.PointerDeviceKind.mouse ? _mouseDeviceId :
-        event.pointerId!.toInt();
+    // All mouse pointer events are given `_mouseDeviceId`, including wheel
+    // events, because wheel events might come before any other PointerEvents,
+    // and wheel PointerEvents don't contain pointerIds.
+    return switch(_pointerTypeToDeviceKind(event.pointerType!)) {
+      ui.PointerDeviceKind.mouse => _mouseDeviceId,
+      ui.PointerDeviceKind.stylus => _stylusDeviceId,
+      _ => event.pointerId!.toInt(),
```

Comment:
I guess `_` case covers the multi-tapping with fingers or in the trackpad?

(Should we make this `switch` exhaustive, for clarity?)

Reply:
The switch case as well as `_` is just a refactor of what was before the PR, except that I added a branch for `stylus`. I don't think it changes how it behaves for trackpad.

Also, AFAIK, trackpad events are handled as wheel events in `_convertWheelEventToPointerData`, which does not call `_getPointerId`, but always use `_mouseDeviceId` instead.